sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/delete_key_operation.hpp

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief A long-running operation for deleting a Key.
+ *
+ */
+
+#pragma once
+
+#include <azure/core/http/http.hpp>
+#include <azure/core/operation.hpp>
+#include <azure/core/operation_status.hpp>
+#include <azure/core/response.hpp>
+
+#include <azure/keyvault/common/internal/keyvault_pipeline.hpp>
+#include <azure/keyvault/common/keyvault_exception.hpp>
+
+#include "azure/keyvault/keys/deleted_key.hpp"
+
+#include <memory>
+#include <string>
+#include <thread>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
+
+  /**
+   * @brief A long running operation to delete a key.
+   *
+   */
+  class DeleteKeyOperation
+      : public Azure::Core::Operation<Azure::Security::KeyVault::Keys::DeletedKey> {
+  private:
+    /* DeleteKeyOperation can be constructed only by friends classes (internal creation). The
+     * constructor is private and requires internal components.*/
+    friend class KeyClient;
+
+    std::shared_ptr<Azure::Security::KeyVault::Common::Internal::KeyVaultPipeline> m_pipeline;
+    Azure::Security::KeyVault::Keys::DeletedKey m_value;
+    std::unique_ptr<Azure::Core::Http::RawResponse> m_rawResponse;
+    std::string m_continuationToken;
+
+    /* This is the implementation for checking the status of a deleted key. The key is considered
+     * deleted if querying /deletedkeys/keyName returns 200 from server. Or whenever soft-delete is
+     * disabled.*/
+    std::unique_ptr<Azure::Core::Http::RawResponse> PollInternal(
+        Azure::Core::Context& context) override;
+
+    Azure::Core::Response<Azure::Security::KeyVault::Keys::DeletedKey> PollUntilDoneInternal(
+        Azure::Core::Context& context,
+        std::chrono::milliseconds period) override
+    {
+      while (!IsDone())
+      {
+        m_rawResponse = Poll(context);
+        std::this_thread::sleep_for(period);
```

Comment:
Reverse the order to sleep before the Poll.  Otherwise we potentially sleep even though the Poll resulted in IsDone.

Other option would be to IsDone() -> Poll() -> IsDone() -> Sleep() -> repeat